raiden/network/channelgraph.py

```diff
@@ -75,16 +167,90 @@ def get_paths_of_length(self, source, num_hops=1):
             if len(path) == num_hops + 1
         ]
 
-    def has_path(self, source, target):
-        """ Return True if there is a path connecting source and target, False
-        otherwise.
+    def get_best_routes(self, our_address_bin, target_address_bin, amount, lock_timeout=None):
+        """ Yield a two-tuple (path, channel) that can be used to mediate the
+        transfer. The result is ordered from the best to worst path.
         """
-        return networkx.has_path(self.graph, source, target)
+        available_paths = self.get_shortest_paths(
+            our_address_bin,
+            target_address_bin,
+        )
+
+        # XXX: consider using multiple channels for a single transfer. Useful
+        # for cases were the `amount` is larger than what is available
+        # individually in any of the channels.
+        #
+        # One possible approach is to _not_ filter these channels based on the
+        # distributable amount, but to sort them based on available balance and
+        # let the task use as many as required to finish the transfer.
+
+        for path in available_paths:
+            partner = path[1]
+            channel = self.partneraddress_channel[partner]
+
+            if not channel.isopen:
+                if log.isEnabledFor(logging.INFO):
+                    log.info(
+                        'channel %s - %s is close, ignoring',
+                        pex(path[0]),
+                        pex(path[1]),
+                    )
+
+                continue
 
-    def add_path(self, from_, to_):
+            # we can't intermediate the transfer if we don't have enough funds
+            if amount > channel.distributable:
+                if log.isEnabledFor(logging.INFO):
+                    log.info(
+                        'channel %s - %s doesnt have enough funds [%s], ignoring',
+                        pex(path[0]),
+                        pex(path[1]),
+                        amount,
+                    )
+                continue
+
+            if lock_timeout:
+                # Our partner wont accept a lock timeout that:
+                # - is larger than the settle timeout, otherwise the lock's
+                # secret could be release /after/ the channel is settled.
```

Comment:
3. Not true, the last block is included in the [`settle_period`](https://github.com/raiden-network/raiden/blob/master/raiden/smart_contracts/NettingChannelLibrary.sol#L58)